src/prpy/serialization.py

```diff
@@ -320,44 +354,83 @@ def deserialize_environment(data, env=None, purge=False, reuse_bodies=None):
 
     return env
 
-def deserialize_kinbody(env, data, name=None, anonymous=False, state=True):
+def deserialize_kinbody(env, data, name=None, anonymous=False, state=True,
+        urdf_module_getter=None):
+
     from openravepy import RaveCreateKinBody, RaveCreateRobot
 
+    if urdf_module_getter is None:
+        urdf_module_getter = UnitaryMemoizer(lambda: openravepy.RaveCreateModule(env,'urdf'))
+
     deserialization_logger.debug('Deserializing %s "%s".',
         'Robot' if data['is_robot'] else 'KinBody',
         data['name']
     )
-
-    link_infos = [
-        deserialize_link_info(link_data['info']) \
-        for link_data in data['links']
-    ]
-    joint_infos = [
-        deserialize_joint_info(joint_data['info']) \
-        for joint_data in data['joints']
-    ]
-
-    if data['is_robot']:
-        # TODO: Also load sensors.
-        manipulator_infos = [
-            deserialize_manipulator_info(manipulator_data['info']) \
-            for manipulator_data in data['manipulators']
-        ]
-        sensor_infos = []
-
-        kinbody = RaveCreateRobot(env, '')
-        kinbody.Init(
-            link_infos, joint_infos,
-            manipulator_infos, sensor_infos,
-            data['uri']
-        )
+    
+    name_desired = name or data['name']
+    
+    if data.get('uri_only', False):
+        
+        if data['is_robot']:
+
+            parts = data['uri'].split()
+            if len(parts)==2 and parts[0].endswith('.urdf') and parts[1].endswith('.srdf'):
+                robot_name = urdf_module_getter().SendCommand('Load {}'.format(data['uri']))
+                if robot_name != name_desired:
+                    raise RuntimeError('error, or_urdf name mismatch!')
```

Comment:
This check is incorrect. `or_urdf` loads the robot with a default name generated from the URDF file. This code should call `kinbody.SetName(name_desired)` to override the name with the serialized value.